htlcswitch/quiescer.go

```diff
@@ -490,6 +526,34 @@ func (q *QuiescerLive) resume() {
 	q.resumeQueue = nil
 }
 
+// startTimeout starts the timeout function that fires if the quiescer remains
+// in a quiesced state for too long. If this function is called multiple times
+// only the last one will have an effect.
+func (q *QuiescerLive) startTimeout() {
+	if q.cfg.onTimeout == nil {
+		return
+	}
+
+	old := q.timeoutTimer
+
+	q.timeoutTimer = time.AfterFunc(q.cfg.timeoutDuration, q.cfg.onTimeout)
+
+	if old != nil {
```

Comment:
When do we hit the case when we have a non-nil old timer?

Reply:
Yeah however, I think it's still a good defensive move. We have limited ability to rule cases out at compile time, and so even if this is dead code in practice it should make it hold up a bit better in code churn, I think.